tests/plotting/test_widgets.py

```diff
@@ -835,3 +841,163 @@ def test_clear_camera3d_widget(verify_image_cache):
     pl.add_camera3d_widget()
     pl.clear_camera3d_widgets()
     pl.show(cpos='xy')
+
+
+class Test_event_parser:
+    """Class to regroup tests for widgets that use the  `_parse_interaction_event()` function"""
```

Comment:
A lot these function look very similar. Is it possible to parametrize the tests instead? Though I can see there are small differences in the fixtures, mock setup, kwargs passed to widget method, etc., and parametrizing could be a bit complex....

I think it's ok as-is if you think it's not really feasible to parametrize.

Reply:
Thank you for the feedback.
Done in https://github.com/pyvista/pyvista/commit/69a1458a216c77abaf98b4f128f823c211c29b1d with `pytest parametrize`, although some readability was lost on the road :angel: .

FWIW, when it comes to write complex test parametrization, I really like the plugin [pytest_cases](https://smarie.github.io/python-pytest-cases/). I managed to do without it here but it is really powerful.

Reply:
> Done in [69a1458](https://github.com/pyvista/pyvista/commit/69a1458a216c77abaf98b4f128f823c211c29b1d) with `pytest parametrize`, although some readability was lost on the road 👼 .

Thanks, yes it's hard to find a balance with readability and generalization, I think it reads OK here.

> FWIW, when it comes to writing complex test parametrization, I really like the plugin [pytest_cases](https://smarie.github.io/python-pytest-cases/). I managed to do without it here but it is really powerful.

Good to know, thanks for the tip!